src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -548,6 +558,41 @@ void MemoryInputBase::bypassAssignState() {
     return;
 }
 
+MemoryInput::MemoryInput(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr ctx)
+    : MemoryInputBase::MemoryInputBase(op, ctx) {
+    auto rvWithSubgraph = ov::as_type_ptr<ov::intel_cpu::ReadValueWithSubgraph>(op);
+    if (rvWithSubgraph) {
+        body = rvWithSubgraph->get_function();
+    }
+}
+
+MemoryInput::MemoryInput(const std::string id,
+                         const std::string& name,
+                         const std::string& type,
+                         const Shape& output_shape,
+                         const ov::element::Type& output_prc,
+                         const GraphContext::CPtr context,
+                         const ov::optional<std::vector<Shape>>& input_shape,
+                         const ov::optional<std::vector<ov::element::Type>>& input_prc,
+                         std::shared_ptr<ov::Model> func,
+                         mode mode)
+    : MemoryInputBase::MemoryInputBase(id,
+                                       name,
+                                       type,
+                                       output_shape,
+                                       output_prc,
+                                       context,
+                                       input_shape,
+                                       input_prc,
+                                       mode) {
+    body = func;
+
+    if (haveSubgraph() && isDynamic) {
+        // Update to MemInpSingleShapeInfer
+        shapeInference = PassThroughShapeInferFactory(body).makeShapeInfer();
```

Comment:
As you said: "The input subgraph being fused into MemoryInput, may produce shapes different from the Parameters attached to MemoryInput". 
This is my reason of adding this shape infer. 
https://github.com/xipingyan/openvino/blob/b621364e611bf7304a2d286fb914d296f06a4ae5/src/plugins/intel_cpu/src/shape_inference/shape_inference_pass_through.hpp#L36-L49

we need to run `m_body->validate_nodes_and_infer_types(); `in order to get final real shape.

Reply:
I know it introduces unnecessary shape infer if there is no reset state. I will try to move them to runtime.

Reply:
hi @maxnick 
Because you suggest to take InternalDynShapeInferFactory to do shape inference, but I found it do nothing(https://github.com/xipingyan/openvino/blob/2bb7f7b62fc794e0e0cccbc2ebd14d761a1f5255/src/plugins/intel_cpu/src/shape_inference/shape_inference_internal_dyn.hpp#L23).

So I just try to remove all shape inference code, I found it works now. I don't know why? maybe latest code refactoring fixed dynamic shape issue.